### PR TITLE
fix: allow addresses to be AddressPayload or string on DraftOrder creation

### PR DESCRIPTION
**Pull request by kasperkristensen**
**What**
- Allows `shipping_address`and `billing_address`be either an `AddressPayload` or `string` as part of the create draft order endpoint props.

**How**
- Implements the same patterns as used for the similar Cart endpoints.

**Why**
The endpoint was using an old pattern for switching between addresses being either ids or payloads, which resulted in `billing_address` not being added during the creation process if it was not passed as an id.

**Also**
- Fixes the wrong payload types being used in js/react clients.

Closes CORE-720

### CODE REVIEW

File: packages/medusa/src/api/routes/admin/draft-orders/create-draft-order.ts
Comment (olivermrbl) [inline packages/medusa/src/api/routes/admin/draft-orders/create-draft-order.ts:126]: todo: Rename data passed to the DraftOrderService
```suggestion
  const draftOrderDataToCreate: DraftOrderCreateProps = { ...rest }
  ...
```
-------------
suggestion: Move it to the service layer and simplify it similar to how we do in the CartService:
```
const shippingAddress = data.shipping_address_id ?? data.shipping_address
const billingAddress = data.billing_address_id ?? data.billing_address
```

**Review by olivermrbl — CHANGES_REQUESTED** [review]
Nice catch!

I believe we should also allow you to pass address ids as payload when updating a draft order since we call `CartService.update` i.e. the payload to `update-draft-order.ts` should be:
```ts
@IsOptional()
  @IsType([AddressPayload, String])
  billing_address?: AddressPayload | string

  @IsOptional()
  @IsType([AddressPayload, String])
  shipping_address?: AddressPayload | string
```

What do you think?

File: packages/medusa/src/api/routes/admin/draft-orders/create-draft-order.ts
Comment (kasperkristensen) [reply packages/medusa/src/api/routes/admin/draft-orders/create-draft-order.ts]: Regarding the suggestion, I can't change how it's passed right now with out changing up the cart props type which I don't think should be part of this PR. 😄

Comment (olivermrbl) [reply packages/medusa/src/api/routes/admin/draft-orders/create-draft-order.ts]: I suggested migrating it to behave exactly like the `update-cart.ts` flow. I stole the above snippet from the `CartService.update`. Am I missing something?

**Comment by olivermrbl** [comment]
Can I get you to merge in latest from `develop` and resolve conflicts?

File: integration-tests/api/__tests__/admin/draft-order.js
Comment (olivermrbl) [inline integration-tests/api/__tests__/admin/draft-order.js:74]: suggestion(if-minor): We could add a couple of assertions on the actual shipping address + billing address in both of these tests to be 100% sure we've corrected the issue and to avoid future bugs related to it.

File: integration-tests/api/__tests__/admin/draft-order.js
Comment (olivermrbl) [inline integration-tests/api/__tests__/admin/draft-order.js:112]: See other comment. We could assert on the actual billing_address object as well as the shipping address object. Same goes for the test below.

File: integration-tests/api/__tests__/admin/draft-order.js
Comment (kasperkristensen) [reply integration-tests/api/__tests__/admin/draft-order.js:74]: They aren't returned as part of the request (not part of default relations), but I can add a fetch afterwards check it against that.

Comment (olivermrbl) [reply integration-tests/api/__tests__/admin/draft-order.js:74]: Alright, but maybe just asserting on both ids would suffice then?

Comment (olivermrbl) [reply integration-tests/api/__tests__/admin/draft-order.js:74]: Both the `shipping_address_id` and `billing_address_id` for both tests

**Comment by srindom** [comment]
LGTM! Pending @olivermrbl for final approval 👍

**Review by olivermrbl — APPROVED** [review]
LGTM!

**Pull request merged**
